src/Xamarin.Android.Build.Tasks/Tasks/NdkUtils.cs

```diff
@@ -175,6 +175,22 @@ static bool GetNdkToolchainRelease (string androidNdkPath, out string version)
 			return true;
 		}
 
+		static bool GetNdkToolchainSourceProperties (string androidNdkPath, out NdkVersion version)
+		{
+			version = new NdkVersion ();
+			var sourcePropertiesPath = Path.Combine (androidNdkPath, "source.properties");
+			if (!File.Exists (sourcePropertiesPath)) {
+				return false;
+			}
+			var match = Regex.Match (File.ReadAllText (sourcePropertiesPath).Trim (), "(\\d+).(\\d+).(\\d+)");
```

Comment:
I don't like this regex, as it could match nearly anything, and I'm not sure I want to rely on `source.properties` *only ever* containing two lines, one of which only containing one dotted number.

I would instead prefer matching on:

    ^Pkg.Revision\s*=\s*([.0-9]+)$

This way we ensure we *only* grab the value of the `Pkg.Revision` "key".